src/main/java/edu/greenblitz/robotName/subsystems/swerve/Modules/SwerveModule.java

```diff
@@ -51,85 +55,97 @@ public double getCurrentMeters() {
     }
 
     public SwerveModulePosition getCurrentPosition() {
-        return new SwerveModulePosition(getCurrentMeters(), new Rotation2d(getModuleAngle()));
+        return new SwerveModulePosition(getCurrentMeters(), getModuleAngle());
     }
 
-    public void resetEncoderToValue(double angleInRads) {
-        swerveModule.resetAngle(angleInRads);
+    public void resetEncoderToValue(Rotation2d angle) {
+        swerveModule.resetAngle(angle);
     }
 
     public void resetEncoderToValue() {
-        resetEncoderToValue(0);
-    }
-
-    public void periodic() {
-        swerveModule.updateInputs(swerveModuleInputs);
-        Logger.getInstance().processInputs("DriveTrain/Module" + this.module.toString(), swerveModuleInputs);
+        swerveModule.resetAngle(Rotation2d.fromRadians(0));
     }
 
 
     public void resetEncoderByAbsoluteEncoder() {
-        resetEncoderToValue(swerveModuleInputs.absoluteEncoderPosition);
+        resetEncoderToValue(Rotation2d.fromRadians(swerveModuleInputs.absoluteEncoderPosition));
     }
 
-    public void setLinSpeed(double speed) {
+    public void setLinearVelocity(double speed) {
+        targetVelocity = speed;
         swerveModule.setLinearVelocity(speed);
     }
 
     public void stop() {
         swerveModule.stop();
     }
 
-    public SwerveModuleState getModuleState(){
+    public SwerveModuleState getModuleState() {
         return new SwerveModuleState(
                 getCurrentVelocity(),
-                new Rotation2d(getModuleAngle())
+                getModuleAngle()
         );
     }
 
-    public boolean isAtAngle(double targetAngleInRads, double tolerance) {
-        double currentAngleInRads = getModuleAngle();
-        return (currentAngleInRads - targetAngleInRads) %(2*Math.PI) < tolerance
-                || (targetAngleInRads - currentAngleInRads) % (2*Math.PI) < tolerance;
+    public boolean isAtAngle(Rotation2d targetAngle, Rotation2d errorAngleTolerance) {
+
+        if(
```

Comment:
This if is not needed, simply return the value

Reply:
nitzan wanted it simpler and readabler. now it is. do you want me to change that to what is was?

Reply:
```suggestion
        return (getModuleAngle().getRadians() - targetAngle.getRadians()) % (2 * Math.PI) < errorAngleTolerance.getRadians() ||
                (targetAngle.getRadians() - getModuleAngle().getRadians()) % (2 * Math.PI) < errorAngleTolerance.getRadians();
```

Reply:
i done that but it dont let me resolve.